packages/block-editor/src/store/selectors.js

```diff
@@ -1628,8 +1628,18 @@ const isBlockVisibleInTheInserter = (
 	checkedBlocks.add( blockName );
 
 	// If parent blocks are not visible, child blocks should be hidden too.
-	if ( !! blockType.parent?.length ) {
-		return blockType.parent.some(
+	//
+	// In some scenarios, blockType.parent may be a string.
+	// A better approach would be sanitize parent in all the places that can be modified:
+	// block registration, processBlockType, filters, etc.
+	// In the meantime, this is a hotfix to prevent the editor from crashing.
+	const parent =
+		typeof blockType.parent === 'string' ||
+		blockType.parent instanceof String
+			? [ blockType.parent ]
+			: blockType.parent;
+	if ( Array.isArray( parent ) ) {
```

Comment:
Is this condition still necessary. I guess right now it's more `if ( !! parent )`

Reply:
I'm curious why we'd need both `typeof blockType.parent === 'string' || blockType.parent instanceof String`.. 🤔

Reply:
> I'm curious why we'd need both typeof blockType.parent === 'string' || blockType.parent instanceof String.. 

Otherwise, we don't cover all potential issues. Note the issue was that we checked for `.length` in a type that wasn't an array (didn't have `.some`). That's the case for both literal & object strings.